File: CarAuction.Backend/CarAuction.Domain/Entities/Listing.cs
namespace CarAuction.Domain.Entities;

public class Listing
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public decimal Price { get; set; }
    public bool IsAuction { get; set; }
    public string Status { get; set; } = "active";
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }

    // Navigation properties
    public User? User { get; set; }
    public Auction? Auction { get; set; }
    public ICollection<Image> Images { get; set; } = new List<Image>();
}

File: CarAuction.Frontend/src/types/listing.ts
import { Image } from './common';

export interface Listing {
  id: number;
  userId: number;
  title: string;
  description: string;
  price: number;
  isAuction: boolean;
  status: 'active' | 'sold' | 'expired' | 'cancelled';
  createdAt: string;
  updatedAt?: string;
  images?: Image[];
}

export interface CreateListingRequest {
  title: string;
  description: string;
  price: number;
  isAuction: boolean;
  startingPrice?: number;
  startTime?: string;
  endTime?: string;
  minBidIncrement?: number;
}

export interface UpdateListingRequest {
  title: string;
  description: string;
  price: number;
}

export interface ListingFilters {
  search?: string;
  minPrice?: number;
  maxPrice?: number;
  status?: string;
  isAuction?: boolean;
  page: number;
  limit: number;
}
